rfcs/text/0049-entity-fields.md

```diff
@@ -20,31 +21,55 @@ This approach would allow ECS to accommodate new types of entities without requi
 
-- Host entities should utilize the `host.*` field set to capture detailed host information
-- User entities should leverage the `user.*` field set for user-specific attributes
-- Cloud resource entities should use the `cloud.*` field set for cloud provider details
-- Service entities should employ the `service.*` field set for service-specific metadata
+- Host entities should also utilize the `host.*` field set to capture detailed host information
+- User entities should also leverage the `user.*` field set for user-specific attributes
+- Cloud resource entities should also use the `cloud.*` field set for cloud provider details
+- Service entities should also employ the `service.*` field set for service-specific metadata
 
-This approach ensures backward compatibility, maintains existing ECS patterns, and preserves the rich metadata capabilities of established field sets while allowing the entity fields to provide a consistent way to identify and categorize all entity types. The entity fields serve as a complementary layer that enables unified entity representation, particularly for entity types that don't have dedicated field sets.
+This approach ensures backward compatibility, maintains existing ECS patterns, and preserves the rich metadata capabilities of established field sets while allowing the entity fields to provide a consistent way to identify and categorize *all* entity types. The entity fields serve as a complementary layer that enables unified entity representation, particularly for entity types that don't have dedicated field sets.
 
-## Usage
+## Field Re-use
 
-The entity field set enables us to normalize entity data in such a way where we can easily query key attributes in a standardized way regardless of the type and source of the entity. This will be how we'll normalize all entity data in the upcoming inventory experience that we're planning for the security solution.
+### ECS Producers
+
+For ECS producers, such as Beats, Elastic Agent integrations, ingest pipelines, and other methods for shipping data to Elastic, the `entity.*` fields are expected to be nested as follows:
+- If the entity type is one of host, user, service, cloud, orchestrator), then the entity fields should be nested under the respecitve root field set, for example `host.entity.*` , `user.entity.*`, etc.
+- If the entity type is not one of the above, then that `entity.*` fields should be nested under a new root-level object, called `generic`, as `generic.entity.*`
+
+Special note for use with entity fields that use `target` fields.
+If the entity type includes a provision for `target.*` fields, such as `user.target.*`, then the `entity.*` fields should be nested under the `*.target.*` field set, for example, `user.target.entity.*`.
+
+### ECS Consumers or Data Stores
+
+For ECS consumers, such as the Elastic Security Solution entity store indices, the `entity.*` fields should be used directly at the root of the events.
```

Comment:
Could you expand on what this means, I don't think I understand it. What do you mean by "used". Why can a data store only use `entity.*` if the producers can write to other top level fieldsets?

I think this is also an implementation specific detail that doesn't need to be part of ECS, so maybe it can be removed.

Reply:
I agree we can remove this from the ECS spec, but the idea is that the entity store schema will include entity.* at the root level in each document.  this will allow us and users to query the entity store for entity ID's (and other attributes) w/o burdening them with knowing under which entity class {host, user, service, generic, etc.} it might be stored.